### PR TITLE
test(e2e): add canister ids as env vars for e2e tests

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to access canister_e2e_ids.json in e2e tests as env vars.

### CODE REVIEW

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM!

File: tsconfig.eslint.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
 		"./src/**/*.js",
 		"./src/**/*.ts",
 		"./src/**/*.svelte",
+		"./shared.utils.ts",
```
Comment (peterpeterparker) [inline tsconfig.eslint.json:12]: `shared.utils` feels a bit like reading `utils.utils` 😄. What about `env.utils.ts` or `canister-ids.utils.ts`?

File: playwright.config.ts [hunk 2]
```diff
@@ -1,4 +1,15 @@
 import { defineConfig, devices } from '@playwright/test';
+import dotenv, { type DotenvPopulateInput } from 'dotenv';
+import { join } from 'node:path';
+import { readCanisterIdsFromJSONFile } from './shared.utils';
+
+dotenv.populate(
+	process.env as DotenvPopulateInput,
```
Comment (peterpeterparker) [inline playwright.config.ts:7]: the type cast is a must?

File: shared.utils.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { readFileSync } from 'node:fs';
+
+/**
+ * Read a JSON file with canister IDs and transform it into a dictionary. Optionally add prefix to all dictionary keys.
+ */
+export const readCanisterIdsFromJSONFile = ({
```
Comment (peterpeterparker) [inline shared.utils.ts:6]: nth: can be shortened `readCanisterIds`. From IC dev perspective, it's the standard to read ids from a file

File: shared.utils.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	filePath,
+	prefix
+}: {
+	filePath: string;
+	prefix?: string;
+}): Record<string, string> => {
+	try {
```
Comment (peterpeterparker) [inline shared.utils.ts:13]: Nothing changed in the logic of this function correct? Simply refactoring and it does need a review?

File: tsconfig.eslint.json (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply tsconfig.eslint.json:12]: `env.utils.ts` it is 👍

File: shared.utils.ts (hunk 3 above)
Comment (DenysKarmazynDFINITY) [reply shared.utils.ts:6]: Agreed.

File: shared.utils.ts (hunk 4 above)
Comment (DenysKarmazynDFINITY) [reply shared.utils.ts:13]: Yep, I only extracted this function into a separate file.

File: playwright.config.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply playwright.config.ts:7]: This cast is needed to align global env type (where each value is `string | undefined`) with the dotenv.populate function param (`DotenvPopulateInput`, where each value is `string`). It should be safe to cast, as even if some var doesn't have a value in an env file, it still resolves to an empty string in `process.env`.

Comment (peterpeterparker) [reply playwright.config.ts:7]: Alright. Thanks for the feedback.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**